src/parsers.rs

```diff
@@ -0,0 +1,171 @@
+//! Utilities for parsing the contents of different file types into a text
+//! representation that can be sent to the LanguageTool API.
+
+/// Parse the contents of an HTML file into a text format to be sent to the
+/// LanguageTool API.
+#[cfg(feature = "html")]
+pub fn parse_html(file_content: impl AsRef<str>) -> String {
+    use html_parser::Node;
+
+    let mut txt = String::new();
+
+    let html = html_parser::Dom::parse(file_content.as_ref()).unwrap_or_default();
+    let mut children: Vec<Node> = html.children.into_iter().rev().collect();
+
+    fn handle_node(txt: &mut String, node: Node) {
+        if let Some(e) = node.element() {
+            match e.name.as_str() {
+                "head" | "script" | "style" => {
+                    return;
+                },
+                "code" => {
+                    txt.push_str("_code_");
+                    return;
+                },
+                "a" => {
+                    txt.push_str("_link_");
+                    return;
+                },
+                "pre" => {
+                    txt.push_str("_pre_");
+                    txt.push_str("\n\n");
```

Comment:
It is a bit linked to my general file-comment, but there you could simply return a `Markup` note, with no `interpret_as`, and LT should ignore it, I think.